quarkchain/cluster/tests/test_shard_state.py

```diff
@@ -2751,6 +2752,62 @@ def test_enable_evm_timestamp_with_contract_call(self):
         ):
             state.finalize_and_add_block(b1)
 
+    def test_enable_evm_timestamp_qkchash_with_rotation_stats(self):
+        id1 = Identity.create_random_identity()
+        acc1 = Address.create_from_identity(id1, full_shard_key=0)
+
+        env = get_test_env(genesis_account=acc1, genesis_minor_quarkash=10000000)
+        state = create_default_shard_state(env=env)
+        state.shard_config.CONSENSUS_TYPE = ConsensusType.POW_QKCHASH
+        # set the initial enabled Qkchashx block height to one
+        state.env.quark_chain_config.ENABLE_QKCHASHX_HEIGHT = 1
+
+        # generate a minor block to mine
+        b1 = state.get_tip().create_block_to_append(address=acc1, difficulty=5)
```

Comment:
line 2766 to line 2778 can be generalized as a function and reused for line 2791 to 2802